torch/lib/THD/CMakeLists.txt

```diff
@@ -140,6 +147,16 @@ ENDIF()
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 
 ADD_LIBRARY(THD STATIC ${all_cpp})
+
+if (CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO)
+  ADD_DEPENDENCIES(THD caffe2_library)
```

Comment:
if you ADD_DEPENDENCIES on `caffe2` (instead of `caffe2_library`), it should include the caffe2 directories (which include the dirs below) automatically.
That's the right thing to do, because Caffe2's header paths might keep changing, and you're effectively manually maintaining the paths below